File: -Parallel-computing-processes-Lr_1-CSharp-main/Lr_1/SumThread.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lr_1
{
    public class SumThread
    {
        private int id;
        private StopThread stopThread;
        private int step;

        public SumThread(int id, StopThread stopThread, int step)
        {
            this.id = id;
            this.stopThread = stopThread;
            this.step = step;
        }

        public void Run()
        {
            long sum = 0;
            int additions = 0;
            bool isStop = false;
            do
            {
                sum+= additions * step;
                additions++;
                isStop = stopThread.IsCanStop();
            } while (!isStop);
            Console.WriteLine($"Thread {id} - Sum: {sum} - Additions: {additions}");
        }
    }
}

File: -Parallel-computing-processes-Lr_1-Java-main/src/SumThread.java
public class SumThread extends Thread{
    private final int id;
    private final StopThread stopThread;
    private final int step;

    public SumThread(int id, StopThread stopThread, int step) {
        this.id = id;
        this.stopThread = stopThread;
        this.step=step;
    }

    @Override
    public void run() {
        long sum = 0;
        int additions = 0;
        boolean isStop = false;
        do{
            sum+=additions*step;
            additions++;
            isStop = stopThread.isCanStop();
        } while (!isStop);
        System.out.println("Thread " + id + " - Sum: " + sum + " - Additions: " + additions);
    }
}
